### PR TITLE
docs: OpenID Connect under federated login.

### PR DESCRIPTION
**Pull request by merlin-northern**
Changelog: Title
Ticket: MEN-7075

### CODE REVIEW

File: 08.Server-integration/09.OIDC-Federated-Authentication/docs.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+---
+
+!!!!! OpenID Connect Federated Authentication is only available in the Mender Enterprise plan.
+!!!!! See [the Mender features page](https://mender.io/product/features?target=_blank)
+!!!!! for an overview of all Mender plans and features.
+
+OpenID Connect (OIDC) uses a very similar flow to [SAML](../08.SAML-Federated-Authentication).
+And as with SAML, you can configure Mender to initiate the authorisation process
+with the OpenID Connect provider by exposing a login endpoint which redirects
+user to the authorisation server which in turn calls authentication callback.
```
Comment (tranchitella) [inline 08.Server-integration/09.OIDC-Federated-Authentication/docs.md:14]:
```suggestion
OpenID Connect (OIDC) uses a very similar flow to [SAML](../08.SAML-Federated-Authentication).
And as with SAML, you can configure Mender to initiate the authorization process
with the OpenID Connect provider by exposing a login endpoint that redirects
user to the authorization server, which in turn calls an authentication callback.
```

File: 08.Server-integration/09.OIDC-Federated-Authentication/docs.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+Please note that currently Mender implements [Implicit Flow](https://openid.net/specs/openid-connect-implicit-1_0.html).
```
Comment (tranchitella) [inline 08.Server-integration/09.OIDC-Federated-Authentication/docs.md:16]:
```suggestion
Please note that as of today, Mender only implements the [Implicit Flow](https://openid.net/specs/openid-connect-implicit-1_0.html).
```

File: 08.Server-integration/09.OIDC-Federated-Authentication/docs.md [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+to initiate and complete the OIDC flow.
+
+### The well-known endpoint
+
+There is a number of parameters needed for OIDC to operate. You can get all of them from the so-called
+_well-known_ endpoint, and use those to create a provider, or pass the well-known url in the creation
+process. In the latter case you need to use the following document (either
+in the [organization-and-billing](https://hosted.mender.io/ui/settings/organization-and-billing)
+or in the POST API call to [https://hosted.mender.io/api/management/v1/useradm/sso/idp/metadata](https://docs.mender.io/api/#management-api-user-administration-and-authentication-post-saml--openid-connect-metadata--or-url-point-to-them)
+endpoint)
```
Comment (tranchitella) [inline 08.Server-integration/09.OIDC-Federated-Authentication/docs.md:32]:
```suggestion
There are several parameters needed for OIDC to operate. You can get all of them from the so-called
_well-known_ endpoint, and use those to create a provider or pass the well-known URL in the creation
process. In the latter case, you need to use the following document (either
in the [organization-and-billing](https://hosted.mender.io/ui/settings/organization-and-billing)
or in the POST API call to [https://hosted.mender.io/api/management/v1/useradm/sso/idp/metadata](https://docs.mender.io/api/#management-api-user-administration-and-authentication-post-saml--openid-connect-metadata--or-url-point-to-them)
endpoint)
```

File: 08.Server-integration/09.OIDC-Federated-Authentication/docs.md [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+
+### Upload the metadata and get the login url
```
Comment (tranchitella) [inline 08.Server-integration/09.OIDC-Federated-Authentication/docs.md:34]:
```suggestion
### Upload the metadata and get the login URL
```

File: 08.Server-integration/09.OIDC-Federated-Authentication/docs.md [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+
+In order to create an OIDC provider in Mender you can use the UI
+[organization-and-billing](https://hosted.mender.io/ui/settings/organization-and-billing).
+Enabling the Single Sign-On and choosing the type will bring you the possibility to type or upload
+the metadata you need to configure OpenID Connect as a way to authenticate users.
```
Comment (tranchitella) [inline 08.Server-integration/09.OIDC-Federated-Authentication/docs.md:39]:
```suggestion
To create an OIDC provider in Mender, you can use the UI accessing the [Organization and billing](https://hosted.mender.io/ui/settings/organization-and-billing) view.
Enabling the Single Sign-On and choosing the type will allow you to type or upload the metadata you need to configure OpenID Connect to authenticate users.
```

File: 08.Server-integration/09.OIDC-Federated-Authentication/docs.md [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+
+Mender can use the well-known url, and in that case it is enough to provide the following
+document:
```
Comment (tranchitella) [inline 08.Server-integration/09.OIDC-Federated-Authentication/docs.md:42]:
```suggestion
Mender can use the well-known URL, and in that case, it is enough to provide the following
document:
```

File: 08.Server-integration/09.OIDC-Federated-Authentication/docs.md [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+  "name":"identity-provider-name",
+  "client_id":"Ylb6gfnmXckxHFoXH1aKLX2poDCvS9MV",
+  "client_secret":"zygGHMTL9VlCQpOIHNbXpTZqd77LqqIP",
+  "well_known_url":"https://your.openidconnect.provider.oidc/.well-known/openid-configuration"
+}
+```
+
+In the above the name is obligatory while both `client_id` and `client_secret` are optional, but maybe required
+by your OpenID Connect provider. The `well_known_url` denotes the URL for the well-known endpoint which contains
+the hostname of your OIDC provider (called "your.openidconnect.provider.oidc" throughout this section).
```
Comment (tranchitella) [inline 08.Server-integration/09.OIDC-Federated-Authentication/docs.md:55]:
```suggestion
In the above, the name is mandatory, while both `client_id` and `client_secret` are optional but may be required by your OpenID Connect provider. The `well_known_url` denotes the URL for the well-known endpoint, which contains the hostname of your OIDC provider (called "your.openidconnect.provider.oidc" throughout this section).
```

File: 08.Server-integration/09.OIDC-Federated-Authentication/docs.md [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+    ],
+    "request_uri_parameter_supported": false,
+    "request_parameter_supported": false,
+    "token_endpoint_auth_signing_alg_values_supported": [
+      "RS256"
+    ],
+    "end_session_endpoint": "https://your.openidconnect.provider.oidc/oidc/logout"
+  }
+}
+}
```
Comment (tranchitella) [inline 08.Server-integration/09.OIDC-Federated-Authentication/docs.md:147]: Extra closed parenthesis

```suggestion
```

File: 08.Server-integration/09.OIDC-Federated-Authentication/docs.md [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+```
+
+In the former case, Mender will query and validate the required settings.
+Once you see the success on parsing and saving the metadata the UI will
+present you with a start or login url: this is the address you use
+to initiate the login process and this completes the setup.
```
Comment (tranchitella) [inline 08.Server-integration/09.OIDC-Federated-Authentication/docs.md:153]:
```suggestion
In the first case, Mender will query and validate the required settings.
Once you see the success of parsing and saving the metadata, the UI will
present you with a start or login URL: this is the address you use
to initiate the login process, which completes the setup.
```

**Review by tranchitella — APPROVED** [review]
I approve leaving a few suggestions/comments, feel free to ignore.

File: 08.Server-integration/09.OIDC-Federated-Authentication/docs.md (hunk 1 above)
Comment (merlin-northern) [reply 08.Server-integration/09.OIDC-Federated-Authentication/docs.md]: fixed.

File: 08.Server-integration/09.OIDC-Federated-Authentication/docs.md (hunk 2 above)
Comment (merlin-northern) [reply 08.Server-integration/09.OIDC-Federated-Authentication/docs.md]: fixed.

File: 08.Server-integration/09.OIDC-Federated-Authentication/docs.md (hunk 3 above)
Comment (merlin-northern) [reply 08.Server-integration/09.OIDC-Federated-Authentication/docs.md]: fixed.

File: 08.Server-integration/09.OIDC-Federated-Authentication/docs.md (hunk 4 above)
Comment (merlin-northern) [reply 08.Server-integration/09.OIDC-Federated-Authentication/docs.md]: fixed.

File: 08.Server-integration/09.OIDC-Federated-Authentication/docs.md (hunk 5 above)
Comment (merlin-northern) [reply 08.Server-integration/09.OIDC-Federated-Authentication/docs.md]: fixed.

File: 08.Server-integration/09.OIDC-Federated-Authentication/docs.md (hunk 6 above)
Comment (merlin-northern) [reply 08.Server-integration/09.OIDC-Federated-Authentication/docs.md]: fixed.

File: 08.Server-integration/09.OIDC-Federated-Authentication/docs.md (hunk 7 above)
Comment (merlin-northern) [reply 08.Server-integration/09.OIDC-Federated-Authentication/docs.md]: fixed.

File: 08.Server-integration/09.OIDC-Federated-Authentication/docs.md (hunk 8 above)
Comment (merlin-northern) [reply 08.Server-integration/09.OIDC-Federated-Authentication/docs.md]: fixed. nicely caught!

File: 08.Server-integration/09.OIDC-Federated-Authentication/docs.md (hunk 9 above)
Comment (merlin-northern) [reply 08.Server-integration/09.OIDC-Federated-Authentication/docs.md]: fixed.

**Pull request merged**